samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImplGraph.java

```diff
@@ -166,6 +166,11 @@ OperatorImpl createAndRegisterOperatorImpl(OperatorSpec prevOperatorSpec, Operat
       operatorImpl.init(config, context);
       operatorImpl.registerInputStream(inputStream);
 
+      if (operatorSpec.getTimerFn() != null) {
+        final TimerRegistry timerRegistry = operatorImpl.createTimerRegistry();
+        operatorSpec.getTimerFn().registerTimer(timerRegistry);
```

Comment:
Same question as before, can we do this in OperatorImpl.init?

Also, it's confusing that operatorImpl has a createTimerRegistry, since the TimerRegistry has already been created by the container and passed to the StreamOperatorTask. Maybe rename to createOperatorTimerRegistry to disambiguate.

Reply:
The base OperatorImpl doesn't have operatorSpec, so I am not able to move the logic there.

Fixed the naming of the method.